staging/src/k8s.io/component-base/logs/logs.go

```diff
@@ -76,3 +82,82 @@ func GlogSetter(val string) (string, error) {
 	}
 	return fmt.Sprintf("successfully set klog.logging.verbosity to %s", val), nil
 }
+
+const logLoggingFormat = "logging-format"
+
+var logRegistry *LogFormatRegistry = NewLogFormatRegistry()
+
+type LogFormatRegistry struct {
+	registry map[string]*logr.Logger
+	mu       sync.Mutex
+}
+
+func NewLogFormatRegistry() *LogFormatRegistry {
+	return &LogFormatRegistry{
+		registry: make(map[string]*logr.Logger),
+		mu:       sync.Mutex{},
+	}
+}
+
+// Register new log format registry to global logRegistry
+func (lfr *LogFormatRegistry) Register(name string, logger logr.Logger) error {
+	if _, ok := lfr.registry[name]; ok {
+		return fmt.Errorf("log format: %s already exists", name)
+	}
+	lfr.mu.Lock()
+	lfr.registry[name] = &logger
+	lfr.mu.Unlock()
+	return nil
+}
+
+// Get specified log format logger
+func (lfr *LogFormatRegistry) Get(name string) (logr.Logger, error) {
+	re, ok := lfr.registry[name]
+	if !ok {
+		return nil, fmt.Errorf("log format: %s does not exists", name)
+	}
+	return *re, nil
+}
+
+// Set specified log format logger
+func (lfr *LogFormatRegistry) Set(name string, logger logr.Logger) {
+	lfr.mu.Lock()
+	lfr.registry[name] = &logger
+	lfr.mu.Unlock()
+}
+
+// Delete specified log format logger
+func (lfr *LogFormatRegistry) Delete(name string) {
+	lfr.mu.Lock()
+	delete(lfr.registry, name)
+	lfr.mu.Unlock()
+}
+
+type LoggingConfig struct {
+	Logr logr.Logger
+}
+
+// NewLoggingConfig return new LoggingConfig
+func NewLoggingConfig(l logr.Logger) *LoggingConfig {
+	return &LoggingConfig{
+		Logr: l,
+	}
+}
+
+// Complete set klog logger
+func (l *LoggingConfig) Complete() {
+	klog.SetLogger(l.Logr)
+}
+
+func initLogFormat() {
```

Comment:
Instead of using init function, please follow convention used for kubernetes genericoptions.

Rename LoggingConfig to Options and implement methods:
* func (o *Options) Validate() []error
* func (o *Options) AddFlags(fs *pflag.FlagSet)
* func (o *Options) Apply()

Example: https://github.com/kubernetes/kubernetes/blob/aa7b71abbd80a19d5d8f3e3a0bed2f605bb38a50/staging/src/k8s.io/component-base/metrics/options.go